sklearn/feature_extraction/text.py

```diff
@@ -567,6 +567,10 @@ def _get_hasher(self):
                              alternate_sign=self.alternate_sign,
                              non_negative=self.non_negative)
 
+    def _get_tags(self):
+        return _update_tags(super(HashingVectorizer, self),
+                            input_types=["string"])
```

Comment:
I have not followed all the earlier discussion on estimator tags, so it may be a stupid question, but it seems a lot of changes are of the form,

```py
class Estimator(BaseEstimator..):
  
    def _get_tags(self):
        return _update_tags(super(Estimator, self),
                            **tags)
```

where,
```py
def _update_tags(sup, **kwargs):
    if hasattr(sup, "_get_tags"):
        tags_old = sup._get_tags().copy()
        tags_old.update(kwargs)
        return tags_old
    else:
        return kwargs.copy()
```

To avoid this method definition in each estimator (which will be a bit tedious to maintain and modify if something changes in the future), couldn't one define a method in the base class inherited everywhere that would iterate over the base classes and retrieve their tags. Very roughly,
```py
import inspect


class BaseEstimator(object):
    tags = {'a': 1}

    def _get_tags(self):
        tags = {}
        tags.update(getattr(self, 'tags', ()))
        for base_class in inspect.getmro(self.__class__):
            if hasattr(base_class, '_get_tags') and base_class != self.__class__:
                tags.update(base_class()._get_tags())
        return tags


class LinearRegression(BaseEstimator):
    tags = {'b': 2}


class LinearRegressionCV(LinearRegression):
    tags = {'c': 4}
```
which would yield,
```py
>>> BaseEstimator()._get_tags()
{'a': 1}
>>> LinearRegression()._get_tags()
{'a': 1, 'b': 2}
>>> LinearRegressionCV()._get_tags()
{'c': 4, 'a': 1, 'b': 2}
```
Do you think there is a reason something like this couldn't work? Or am I missing the point?

(one of the issue with this is that one has to instantiate base classes when querying tags, but I wonder if there is a way around it)

Reply:
Maybe this was discussed before, I need to read the comments..

Reply:
Tags can be decided at runtime is the issue, I think. The current solution is not pretty, happy to revisit it.

Reply:
didn't we expand on this? I can't find the comments any more.
I think the solution was to have your class walking and then define ``_more_tags`` in the estimator, which could or could not be an attribute?

Reply:
I think this solution actually runs into the exact same problem with the MRO as I did initially. This doesn't work as expected, right?